clang/runtime/dpct-rt/include/device.hpp.inc

```diff
@@ -1060,6 +1054,26 @@ static inline sycl::queue &get_default_queue() {
 }
 // DPCT_LABEL_END
 
+// DPCT_LABEL_BEGIN|typedef_queue_ptr|dpct
+// DPCT_DEPENDENCY_BEGIN
+// Device|get_default_queue
+// DPCT_DEPENDENCY_END
+// DPCT_CODE
+
+class queue_ptr {
+  sycl::queue* _q;
+public:
+  queue_ptr(sycl::queue* _p)
+    : _q(_p ? _p : &get_default_queue()) {}
+  queue_ptr()
+    : queue_ptr(0) {}
+  operator sycl::queue*  () const { return _q; }
+  operator sycl::queue*& ()       { return _q; }
+  sycl::queue* operator->() const { return _q; }
```

Comment:
We may need add "operator bool" for this class.

Reply:
With `operator sycl::queue*`, the following conversion sequence implicitly converts a `queue_ptr` to bool: 

`queue_ptr` to `sycl::queue*` (user-defined conversion)
`sycl::queue*` to `bool` (numeric conversion of pointers to bool)